python/packages/autogen-ext/src/autogen_ext/agents/web_surfer/_multimodal_web_surfer.py

```diff
@@ -59,67 +55,122 @@
+        playwright (Playwright, optional): The playwright instance. Defaults to None.
+        context (BrowserContext, optional): The browser context. Defaults to None.
+
+
+
+
+    Example usage:
+
+    The following example demonstrates how to create an video surfing agent with
+    a model client and generate a response to a simple query about a local video
+    called video.mp4.
+
+        .. code-block:: python
+
+
+            import asyncio
+            from autogen_agentchat.ui import Console
+            from autogen_agentchat.teams import RoundRobinGroupChat
+            from autogen_ext.models.openai import OpenAIChatCompletionClient
+            from autogen_ext.agents.web_surfer import MultimodalWebSurfer
 
-class MultimodalWebSurfer(BaseChatAgent):
-    """(In preview) A multimodal agent that acts as a web surfer that can search the web and visit web pages."""
 
-    DEFAULT_DESCRIPTION = "A helpful assistant with access to a web browser. Ask them to perform web searches, open pages, and interact with content (e.g., clicking links, scrolling the viewport, etc., filling in form fields, etc.) It can also summarize the entire page, or answer questions based on the content of the page. It can also be asked to sleep and wait for pages to load, in cases where the pages seem to be taking a while to load."
+            async def main() -> None:
+                # Define an agent
+                web_surfer_agent = MultimodalWebSurfer(
+                    name="MultimodalWebSurfer",
+                    model_client=OpenAIChatCompletionClient(model="gpt-4o-2024-08-06"),
+                )
+
+                # Define a team
+                agent_team = RoundRobinGroupChat([web_surfer_agent], max_turns=3)
+
+                # Run the team and stream messages to the console
+                stream = agent_team.run_stream(task="Navigate to the AutoGen readme on GitHub.")
+                await Console(stream)
+
 
+            asyncio.run(main())
+    """
+
+    DEFAULT_DESCRIPTION = """
+    A helpful assistant with access to a web browser.
+    Ask them to perform web searches, open pages, and interact with content (e.g., clicking links, scrolling the viewport, etc., filling in form fields, etc.).
+    It can also summarize the entire page, or answer questions based on the content of the page.
+    It can also be asked to sleep and wait for pages to load, in cases where the pages seem to be taking a while to load.
+    """
     DEFAULT_START_PAGE = "https://www.bing.com/"
 
+    # Viewport dimensions
+    VIEWPORT_HEIGHT = 900
+    VIEWPORT_WIDTH = 1440
+
+    # Size of the image we send to the MLM
+    # Current values represent a 0.85 scaling to fit within the GPT-4v short-edge constraints (768px)
+    MLM_HEIGHT = 765
+    MLM_WIDTH = 1224
+
+    SCREENSHOT_TOKENS = 1105
+
     def __init__(
         self,
         name: str,
         model_client: ChatCompletionClient,
+        downloads_folder: str | None = None,
```

Comment:
At some point, I think we should have a separate interface/protocol for a web browser so we can enable other browser backends beside playwright.

Reply:
yes we need to define other controller instances, but I don't think it's a priority for now

Reply:
I also think downloads folder is pretty universal. We had this even with the text-based browser before.

Reply:
The reason to allow a None downloads folder is to allow option to disable downloads by default. I want the barebones websurfer to not require any creation of files or logs.